packages/block-editor/src/components/link-control/search-input.js

```diff
@@ -45,21 +45,24 @@ const LinkControlSearchInput = ( {
 	onReset,
 	showInitialSuggestions,
 } ) => {
+	const [ selectedSuggestion, setSelectedSuggestion ] = useState();
+
 	const selectItemHandler = ( selection, suggestion ) => {
 		onChange( selection );
-
-		if ( suggestion ) {
-			onSelect( suggestion );
-		}
+		setSelectedSuggestion( suggestion );
 	};
 
-	const stopFormEventsPropagation = ( event ) => {
+	function selectSuggestionOrCurrentInputValue( event ) {
+		// Avoid default forms behavior, since it's being handled custom here.
 		event.preventDefault();
-		event.stopPropagation();
-	};
+
+		// Interpret the selected value as either the selected suggestion, if
+		// exists, or otherwise the current input value as entered.
+		onSelect( selectedSuggestion || { url: value } );
```

Comment:
I think we need to decide what should be used for `title` here, or at least revise how we're currently treating an empty/unset `title` in the top-level `LinkControl`. Right now I see this issue when entering a raw URL, where the "external link" is empty:

`master`|`fix/link-control-submission`
---|---
![image](https://user-images.githubusercontent.com/1779930/72564034-e4908c80-387c-11ea-8133-b166a3241b2f.png)|![image](https://user-images.githubusercontent.com/1779930/72564083-00942e00-387d-11ea-8f7c-faccf21bd51f.png)

I think `master` is problematic in how it will apply the raw URL as a `title=` attribute of the rendered navigation/button links, but that could be argued to be a separate issue. 

It ties in a bit with #19387 in trying to get some meaningful title from a raw URL, but I think we'd want a sensible fallback for when that can't be retrieved.

Personally, I think it would be expected to omit the `title` (or have a `title: null`) for this fallback of manual/raw URLs. That requires updating the implementation of `LinkControl` to handle this better for its rendering of `ExternalLink`. 

As I write this, I realize this might have been (in part) what @youknowriad was intending in #19462 with the use of `displayURL` here: https://github.com/WordPress/gutenberg/pull/19462/files#diff-106913d5079b57c6032c0775292d8d9dL205

Reply:
Also noting that this is something where having a well-defined `value` shape is important. `title: null` like I mentioned previously may be a semantically-reasonable way to represent an unknown title, but it requires that this is well-understood in how the value is applied. Currently, the blocks do seem to anticipate that this could be empty, but that "empty" would mean _explicitly `undefined`_:

https://github.com/WordPress/gutenberg/blob/669549a36aeba8a3820cb7cd6511845e86d2bf3a/packages/block-library/src/button/edit.js#L104

In the above code, a `title` of `null` would _not_ cause the fallback empty string to be used, and may break the block if it assumes from that point forward that the `title` is guaranteed to be a string.

Reply:
> That requires updating the implementation of `LinkControl` to handle this better for its rendering of `ExternalLink`.

See: #19739